presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -826,6 +838,82 @@ private Global deriveGlobalProperties(TableLayout layout, Map<ColumnHandle, Vari
         return inputToOutput;
     }
 
+    private static Optional<RowExpression> rewriteToCoalesce(
```

Comment:
I will revisit this part later since it looks quite different from https://github.com/prestodb/presto/pull/12586/commits/ee58e47e27dbee15d355e656a3afe9ec60e4420d :)